### PR TITLE
Integration with LaterPay

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/Resources/views/layout.html.twig [hunk 1]
```diff
@@ -20,9 +22,72 @@
                 <a href="{{ path('fos_user_security_login') }}">{{ 'layout.login' | trans({}, 'FOSUserBundle') }}</a>
                 <a href="{{ path('fos_user_registration_register') }}">{{ 'layout.register' | trans({}, 'FOSUserBundle') }}</a>
             {% endif %}
+
+            <div id="laterpay-balance" style="display: inline-block;" class="laterpay-balance"></div>
```
Comment (drdla) [inline app/Resources/views/layout.html.twig]: no inline styles

File: app/Resources/views/layout.html.twig [hunk 2]
```diff
@@ -20,9 +22,72 @@
...
+                            params
+                        );
+
+                        new Y.LaterPay.IFrame(
+                            Y.one("#laterpay-controls"),
+                            lpOptions.controlsLinks,
+                            params
+                        );
+
+                        {# TODO: move to //lpstatic.net/client/X.Y.Z/laterpay.js #}
```
Comment (drdla) [inline app/Resources/views/layout.html.twig]: no TODOs in a pull request

File: src/AppBundle/Listeners/LaterPayRequestListener.php [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            return;
+        }
+
+        if (!$this->manager->isEnabled()) {
+            return;
+        }
+
+        $client = $this->manager->client;
+
+        // TODO: move parameter name to LaterPay Client PHP
```
Comment (drdla) [inline src/AppBundle/Listeners/LaterPayRequestListener.php]: no TODOs in pull requests

File: src/AppBundle/Resources/views/Feed/post.html.twig [hunk 4]
```diff
@@ -1,2 +1,7 @@
-{{ post.truncatedContent }}<br />
-<a href="{{ url('app_post_view', { 'id' : post.id }) }}">Read the full post</a>
+{% if post.price %}
+    {{ post.teaserContent ?: post.truncatedContent }}<br />
```
Comment (drdla) [inline src/AppBundle/Resources/views/Feed/post.html.twig]: no slashes in self-closing tags -> `<br>`

File: src/AppBundle/Resources/views/Post/view.html.twig [hunk 5]
```diff
@@ -4,10 +4,49 @@
...
+                    Read for {{ LaterPay().getPrice(post) }} and pay later
+                {% else %}
+                    Buy the full post with LaterPay for {{ LaterPay().getPrice(post) }}
+                {% endif %}
+            </a>
+
+            <br />
+
+            {% for timepass in timepasses %}
+                <div style="padding: 5px; margin: 2px; display: inline-block; border: 1px solid #E0E0E0;">
```
Comment (drdla) [inline src/AppBundle/Resources/views/Post/view.html.twig]: no inline styles

File: web/css/laterpay.css [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#maincontainer {
```
Comment (drdla) [inline web/css/laterpay.css]: never attach any styles to IDs, only to classes

**Comment by ykasenka-ScnSoft** [comment]
updated

**Comment by doismellburning** [comment]
Closing this 'til we work out what we're doing with this demo!

**Pull request closed without merging**